app/controllers/roles_controller.rb

```diff
@@ -46,8 +46,33 @@ def direct_memberships
     render json: json
   end
   
+  # Find all members of this role.
+  #
+  # For each member, return the full details of the grant as a
+  # JSON object
+  #
+  # +params[:count] returns only the number of members
+  # +params[:limit] and [:offset] control the paging
+  # +params[:search] returns only the members that match the search string
+  def members
+    filter_options, render_options = members_options
+
+    render_options[:order_by] ||= :member_id
+
+    members = @role.memberships_dataset
+                   .search(filter_options)
+                   .select(:role_memberships.*)
+
+    json = render_count_or_results(members, render_options)
```

Comment:
I try to avoid any method name with an "or" or "and" in it.  It's almost always a symptom that it's mixing concerns.  In this particular case, the "error" (as I see it) was in the API design as I mentioned before, but that's just part of the requirement.  That said, we can at least make it as clear as possible that the controller action has two distinct paths.  I also try to avoid `if... else` where I can, but in this case I'd recommend exactly that.  One case for when we're returning a count, another for the normal case.  I really want to draw attention to that bifurcation because it's something I'd consider unexpected and potentially confusing.